pkg/packagekit/updates.jsx

```diff
@@ -449,57 +555,259 @@ class ApplyUpdates extends React.Component {
+            </Expander>
+        </div>
+    </>);
+};
 
+const StatusCard = ({ updates, highestSeverity, timeSinceRefresh, restartableServices, onValueChanged }) => {
+    const numUpdates = Object.keys(updates).length;
+    const numSecurity = count_security_updates(updates);
+    const numRestartServices = restartableServices.daemons.length;
+    const numManualSoftware = restartableServices.manual.length;
+    const numRebootServices = restartableServices.reboot.length;
     let lastChecked;
     if (timeSinceRefresh !== null)
         lastChecked = cockpit.format(_("Last checked: $0"), moment(moment().valueOf() - timeSinceRefresh * 1000).fromNow());
 
-    const icon = highestSeverity ? <span className={PK.getSeverityIcon(highestSeverity)} /> : <CheckIcon color="green" />;
-
-    return (<Flex>
-        <Flex>
-            <FlexItem>{icon}</FlexItem>
-        </Flex>
-        <Flex flex={{ default: 'flex_1' }}>
-            <FlexItem>
-                <Text id="state" component={TextVariants.p}>{stateStr}</Text>
-                <Text id="last-checked" component={TextVariants.small}>{lastChecked}</Text>
-            </FlexItem>
-        </Flex>
-    </Flex>);
+    const notifications = [];
+    if (numUpdates > 0) {
+        if (numUpdates == numSecurity) {
+            const stateStr = cockpit.ngettext("$0 security fix", "$0 security fixes", numSecurity);
+            notifications.push({
+                id: "security-updates-available",
+                stateStr: cockpit.format(stateStr, numSecurity),
+                icon: <span id="icon" className={PK.getSeverityIcon({ info: highestSeverity, iconSize: "fa-lg" })} />,
+                secondary: <Text id="last-checked" component={TextVariants.small}>{lastChecked}</Text>
+            });
+        } else {
+            let stateStr = cockpit.ngettext("$0 update", "$0 updates", numUpdates);
+            if (numSecurity > 0)
+                stateStr += cockpit.ngettext(", including $1 security fix", ", including $1 security fixes", numSecurity);
+            notifications.push({
+                id: "updates-available",
+                stateStr: cockpit.format(stateStr, numUpdates, numSecurity),
+                icon: <span id="icon" className={PK.getSeverityIcon({ info: highestSeverity, iconSize: "fa-lg" })} />,
+                secondary: <Text id="last-checked" component={TextVariants.small}>{lastChecked}</Text>
+            });
+        }
+    } else if (!numRestartServices && !numRebootServices && !numManualSoftware) {
+        notifications.push({
+            id: "system-up-to-date",
+            stateStr: STATE_HEADINGS.uptodate,
+            icon: <CheckIcon color="green" size="md" />,
+            secondary: <Text id="last-checked" component={TextVariants.small}>{lastChecked}</Text>
+        });
+    }
+
+    if (numRebootServices > 0) {
+        const stateStr = cockpit.ngettext("$0 package needs a system reboot", "$0 packages need a system reboot", numRebootServices);
```

Comment:
See above -- please let's find a sensible name for whatever these things are that tracer reports as "needs reboot" (processes, or packages) and then use that consistently.

Reply:
Package fits it the best (the was part of @garrett design to refer to entities which require a system sreboot)